leakcanary/leakcanary-android-release/src/main/java/leakcanary/ScreenOffTrigger.kt

```diff
@@ -42,20 +42,20 @@ class ScreenOffTrigger(
       context: Context,
       intent: Intent
     ) {
-      if (intent.action == ACTION_SCREEN_OFF) {
-        if (currentJob == null) {
-          val job =
-            analysisClient.newJob(JobContext(ScreenOffTrigger::class))
-          currentJob = job
-          analysisExecutor.execute {
+      analysisExecutor.execute {
```

Comment:
If `analysisExecutor.execute` runs the task quickly enough, there's a chance that we might dump the heap & freeze the VM before the process has had a chance to report back to `system_server`, right?

Looks to me like if we want to avoid that, we need to execute with a delay instead"?